karapace/backup/api.py

```diff
@@ -390,14 +378,24 @@ def _handle_restore_topic(
         raise BackupTopicAlreadyExists(f"Topic to restore '{instruction.topic_name}' already exists")
 
 
-def _raise_backup_error(exception: Exception) -> NoReturn:
-    raise BackupDataRestorationError("Error while producing restored messages") from exception
+producer_exception: Exception | None = None
+
+
+def _raise_backup_error(exception: Exception) -> None:
```

Comment:
```suggestion
def _store_backup_error(exception: Exception) -> None:
```

issue: This is no longer raising.